gdal/src/main/scala/geotrellis/raster/gdal/GDALUtils.scala

```diff
@@ -100,8 +100,11 @@ object GDALUtils {
     }
 
   def deriveOverviewStrategyString(strategy: OverviewStrategy): String = strategy match {
+    case Auto(n) if (n == 0) => "AUTO"
     case Auto(n) => s"AUTO-$n"
-    case AutoHigherResolution => "AUTO"
+    case Level(level) => s"$level"
     case Base => "NONE"
+    case other =>
```

Comment:
This would never happen, it is an exhaustive match on a sealed trait.

Reply:
We still have `AutoHigher`, but that has no GDAL string interpretation which makes sense. Is your proposal that it be removed so that this check is once again exhaustive?

Reply:
Ah I forgot about that; :+1: than.

Reply:
Hm, do we need to throw, or warning + a fallback behavior? (default GDALWapr)

Reply:
I've gone with your suggestion of falling back on a sensible default and warning